### PR TITLE
fix uninitialized value and bad free

### PR DESCRIPTION
**Pull request by mikeandmore**
fixed some heap corruption. 

Some of them can be triggered by start/restart fvwm.

### CODE REVIEW

File: fvwm/add_window.c [hunk 1]
```diff
@@ -1819,7 +1819,7 @@ int setup_visible_names(FvwmWindow *fw, int what_changed)
 			&bits, fw, &style, True);
 		if ((changed_names & bits) || (force_update & 2))
 		{
-			free(fw->visible_icon_name);
+			if (fw->visible_icon_name != NoName) free(fw->visible_icon_name);
```
Comment (ThomasAdam) [inline fvwm/add_window.c:1822]: Style-nit:

```
if (fw->visible_icon_name != NoName)
    free(fw->visible_icon_name);
```

File: libs/FEvent.c [hunk 2]
```diff
@@ -897,6 +897,7 @@ Status FGetWMNormalHints(
 {
 	Status ret;
 
+	memset(hints_return, 0, sizeof(XSizeHints));
```
Comment (ThomasAdam) [inline libs/FEvent.c:900]: I doubt this is strictly necessary?

**Review by ThomasAdam — CHANGES_REQUESTED** [review]
Hey @mikeandmore 

Thanks.  One style-nit I'd like you to fix, and a query regarding a `memset()`...

Can you amend the commit message, add:

```
Fixes #446 
```

File: libs/FEvent.c (hunk 2 above)
Comment (mikeandmore) [reply libs/FEvent.c:900]: it is. we kinda ignore the hint masks.

File: fvwm/add_window.c (hunk 1 above)
Comment (mikeandmore) [reply fvwm/add_window.c]: fixed.

**Comment by mikeandmore** [comment]
> ```
> Fixes #446 
> ```

Does it actually fix that? I couldn't reproduce #446. Does this patch work for you?

**Comment by mikeandmore** [comment]
This commit fixes #446

**Pull request merged**